lib/netbox_client_ruby/api/secrets/rsa_key_pair.rb

```diff
@@ -0,0 +1,48 @@
+require 'netbox_client_ruby/entity'
+require 'netbox_client_ruby/api/dcim/device'
+require 'netbox_client_ruby/api/secrets/secret_role'
+
+module NetboxClientRuby
+  module Secrets
+    class RSAKeyPair
+      include Communication
+
+      PATH = '/api/secrets/generate-rsa-key-pair/'.freeze
+
+      def public_key
+        get['public_key']
+      end
+
+      def private_key
+        get['private_key']
+      end
+
+      def reload
+        @response = nil
+      end
+
+      private
+
+      def get
+        if authorization_token
+          @response ||= response connection.get PATH
```

Comment:
I don't like this syntax. I would write
```ruby
@response ||= response connection.get(PATH)
```
Makes it a bit clearer where the arguments belong to.